sklearn/multilabel/classifier_chain.py

```diff
@@ -0,0 +1,134 @@
+            State or seed for random number generator.
+
+        order : list
+            A list of integers specifying the order of the classes in the
+            chain (0 indexed). For example, for a chain of length 5
+                order = [1, 3, 2, 4, 0]
+            means that the first model in the chain will make predictions for
+            column 1 in the Y matrix, the second model will make predictions
+            for column 3, etc. If chain_order is not None it must have a length
+            equal to the number of columns in Y.
+            If chain_order is None an ordered list of integers will be used
+                chain_order = [0, 1, 2, ..., Y.shape[1]]
+            where Y is the label matrix passed to the fit method.
+
+        shuffle : bool
+            If true order is shuffled
+        """
+        self.base_estimator = base_estimator
+        self.random_state = random_state
+        if order is not None:
+            if not isinstance(order, list):
+                order = list(order)
+            if not all([isinstance(i, int) for i in order]):
+                raise ValueError("all elements of order parameter must be "
+                                 "integers")
+
+        self.order = order
+
+        self.shuffle = shuffle
+
+    def fit(self, X, Y):
+        """Fit the ClassifierChain model according to the given training (X)
+        and the true labels (Y) of classes that appear earlier in the chain as
+        defined by the order parameter.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+        Y : array, shape (n_samples, n_classes)
+
+        Attributes
+        ----------
+        classifiers_ : list
+            A list of copies of base_estimator. Once fit the classifiers in
+            this list will be ordered as specified by the order attribute.
+            The original order of labels (as specified by Y) is recovered in
+            the predict method by indexing into the predictions with the
+            list of indices in the order attribute.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+
+        if issparse(Y):
+            Y = Y.toarray()
+
+        random_state = check_random_state(self.random_state)
+        self.classifiers_ = [clone(self.base_estimator)
+                             for _ in range(Y.shape[1])]
+
+        if self.order is not None:
+            if not len(self.order) == Y.shape[1]:
+                raise ValueError("chain_order length must equal n_labels")
+        else:
+            self.order = list(range(Y.shape[1]))
+
+        if self.shuffle:
+            random_state.shuffle(self.order)
+
+        for chain_idx, classifier in enumerate(self.classifiers_):
+            previous_labels = Y[:, self.order[:chain_idx]]
+            y = Y[:, self.order[chain_idx]]
+            X_aug = np.hstack((X, previous_labels))
```

Comment:
As with stacking (see #7427), it makes sense during fitting if we do not use the gold-standard labels, but predictions from jacknifed model, i.e. the output of `cross_val_predict`. Could you add an optional `cv` parameter to support this approach, as in #7427?